app/component/itinerary/ItineraryPage.js

```diff
@@ -995,22 +1021,43 @@ export default function ItineraryPage(props, context) {
     }
   }, [altStates[PLANTYPE.CARTRANSIT][0].plan]);
 
-  // merge the main plan and the scooter plan into one
+  // merge the main plan, the scooter plan and the flex plan into one
   useEffect(() => {
+    let plan = null;
+
     if (
       state.loading === LOADSTATE.DONE &&
       scooterState.loading === LOADSTATE.DONE
     ) {
-      const plan = mergeScooterTransitPlan(
+      plan = mergeScooterTransitPlan(
         scooterState.plan,
         state.plan,
         config.vehicleRental.allowDirectScooterJourneys,
         match.location.query.arriveBy === 'true',
       );
+    }
+
+    if (
+      state.loading === LOADSTATE.DONE &&
+      flexState.loading === LOADSTATE.DONE &&
+      flexState.plan?.edges
+    ) {
+      plan = mergeExternalTransitPlan(
+        flexState.plan,
+        plan,
```

Comment:
I think this will crash if scooterPlan was not already available.

Maybe one can make a single check if all 3 states are in LOADSTATE.DONE, and if so, merge the plans. Otherwise, do nothing.